### PR TITLE
Creating article voting functionality and rendering for each individu…

### PR DESCRIPTION
**Pull request by addoadil**
…al article

### CODE REVIEW

File: apis/votingApiCall.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import axios from "axios";
+
+const addVote = (article_id, flag) => {
```
Comment (popatre) [inline apis/votingApiCall.js]: - Good to see the axios requests extracted into their own folder - nice and reusable

Comment (popatre) [reply apis/votingApiCall.js]: Just thinking of another dev - could rename flag to `addUpVote` - then its either true or false as you pass in

**Review by popatre — CHANGES_REQUESTED** [review]
- Good clear feedback that the vote count has changed
- Good to see the db being updated with the vote increment
- Excellent job handling the votes optimistically - creates a better and quicker UX/UI
- Nice to be able to down vote as well as up
- Excellent work on getting the vote button to only vote once in either direction

- [x] We need some feedback when the vote patch fails for a better UX. At the moment, it looks like the db has been updated successfully when i try to patch with the network disconnected

**Review by HeatherEver — CHANGES_REQUESTED** [review]
The error message looks good for if a user's vote fails but now you're no longer optimistically rendering - we should still be optimistically rendering the vote. If you add a catch block onto your addVote function *inside* the useEffect, in there you can take the vote back off in needed!

**Comment by roisinoreilly** [comment]
really nicely done refactoring this, thank you! PR approved :)

**Pull request merged**